lib/clusterManager.js

```diff
@@ -557,6 +702,35 @@ ClusterManager.prototype.notifyOthers = function (notifyName, args, callback, on
   }
 }
 
+/* Performs a recursive merge on objects, which is different from spread/Object.assign operations 
+that do not merge n > 2 levels down. Does not modify original objects. */
+function recursiveMerge(...objects) {
```

Comment:
Have you tested what happens with objects that have the same key but different value?

Example:

```
let obj1 = {
  key1: {
    key2: 'value',
    key3: 'value'
  }
}
```

```
let obj2  = {
  key1: {
    key2: 'value',
    key4: 'value'
  }
}
```


Will obj 1 no longer have key3 if merged?  Hopefully you tested this

Reply:
Yes, I did, it's in the Sample App PR